app/models/progression.js

```diff
@@ -20,6 +20,10 @@ class Progression {
   getInitialSubset() {
     return this.model.slice(0, 3).map(item => item.id);
   }
+
+  getDateField(id) {
+    return this.model.find(item => item.id === id).dateField;
+  }
```

Comment:
Added hasDateField and incorporated into view which should make update job and any additional date fields easier to accomodate